include/sys/boardctl.h

```diff
@@ -253,6 +261,26 @@ struct boardioc_pm_ctrl_s
 };
 #endif
 
+#ifdef CONFIG_BOARDCTL_NETCONF
+/* Describes the network configuration */
+
+enum boardioc_netconf_e
+{
+  BOARDIOC_NETCONF_DHCP     = 0x1, /* Use DHCP */
+  BOARDIOC_NETCONF_STATIC   = 0x2, /* Use static IP */
+  BOARDIOC_NETCONF_FALLBACK = 0x3, /* Use DHCP with fall back static IP */
+};
+
+struct boardioc_netconf_s
+{
+  enum boardioc_netconf_e flags; /* Configure for static and/or DHCP */
+  uint32_t ipaddr;
+  uint32_t netmask;
+  uint32_t dnsaddr;
+  uint32_t default_router;
+};
+#endif
+
```

Comment:
Most of this is already available in procfs/net/<dev>.  See:

    fs/procfs/fs_procfs.c
    net/procfs/

This is where the NSH ifconfig command gets its network information.

This addition is too redundant and adds no value.  In fact, I say that it degrades the OS overall.  Let's not merge it.

Reply:
If you want to extend the procfs with board-specific info then that might be something better to consider.

Reply:
I believe if we do this we must not use this kludgey approach.  We need to follow the way that Linux does this for compatibility and consistency with some kind of standard:

**10.3. Configuring a DHCP Client**
To configure a DHCP client manually, modify the `/etc/sysconfig/network` file to enable networking and the configuration file for each network device in the `/etc/sysconfig/network-scripts` directory. In this directory, each device should have a configuration file named` ifcfg-eth0`, where `eth0` is the network device name.
The `/etc/sysconfig/network-scripts/ifcfg-eth0` file should contain the following lines:

    DEVICE=eth0
    BOOTPROTO=dhcp
    ONBOOT=yes

A configuration file is needed for each device to be configured to use DHCP.
Other options for the network script includes:

    DHCP_HOSTNAME — Only use this option if the DHCP server requires the client to specify a hostname before receiving an IP address. (The DHCP server daemon in Fedora does not support this feature.)
    PEERDNS=<answer> , where <answer> is one of the following:
        yes — Modify /etc/resolv.conf with information from the server. If using DHCP, then yes is the default.
        no — Do not modify /etc/resolv.conf

https://docs.fedoraproject.org/en-US/Fedora/13/html/Deployment_Guide/s1-dhcp-configuring-client.html

Reply:
We can add the file IO in the netinit as an option. But what do we do in a system with no file system?

Reply:
> We can add the file IO in the netinit as an option. But what do we do in a system with no file system?

Those systems will be unable to use the polled system with DHCP.  Simple.

Reply:
I will mark this a draft so that it is not merged.